fiat/prices.go

```diff
@@ -37,6 +43,44 @@ type PriceSource struct {
 	impl fiatBackend
 }
 
+// PriceSourceConfig is a struct holding various config options used for
+// initialising a new PriceSource.
+type PriceSourceConfig struct {
+	// Backend is the PriceBackend to be used for fetching price data.
+	Backend PriceBackend
+
+	// Granularity specifies the level of granularity with which we want to
+	// get fiat prices. This option is only used for the CoinCap
+	// PriceBackend.
+	Granularity *Granularity
+
+	// PricePoints is a set of price points that is used for fiat related
+	// queries if the PriceBackend being used is the CustomPriceBackend.
+	PricePoints []*Price
+}
+
+// validatePriceSourceConfig checks that the PriceSourceConfig fields are valid
+// given the chosen price backend.
+func validatePriceSourceConfig(cfg *PriceSourceConfig) error {
```

Comment:
could also be `func (p *PriceSourceConfig) validate()error` and then we only validate if non-nil.